rasa/shared/core/slots.py

```diff
@@ -343,9 +343,21 @@ def __init__(
         super().__init__(
             name, initial_value, value_reset_delay, auto_fill, influence_conversation
         )
-        self.values = [str(v).lower() for v in values] if values else []
+        if values and None in values:
+            rasa.shared.utils.io.raise_warning(
+                f"Categorical slot '{self.name}' has `null` listed as a possible value,"
+                f" which translates to `None` in Python. This value is reserved for"
+                f" when a slot is not set and should not be used as a normal value."
+                f" Rasa will proceed, not treating `null` as a possible normal value"
+                f" for the '{self.name}' slot.",
```

Comment:
```suggestion
                f"Categorical slot '{self.name}' has `null` listed as a possible value,"
                f" which translates to `None` in Python. This value is reserved for"
                f" when a slot is not set and should not be specified as a value in the slot definition."
                f" Rasa will ignore `null` as a possible value for the '{self.name}' slot."
                f" Consider renaming this value, for example, `unset`, or explicitly specifying "
                f" the string by using quotation marks, `\"null\"`.",
```

`normal value` is not defined anywhere and might be confusing to the average user. I reworded to avoid use of that phrase. I also added a suggestion at the end for how they could fix this warning.